tests/unit/io/Cereal_unittest.cc

```diff
@@ -0,0 +1,52 @@
+#include <shogun/lib/common.h>
+#include <shogun/base/Parameter.h>
+#include <gtest/gtest.h>
+
+#include <shogun/lib/SGVector.h>
+#include <cereal/archives/json.hpp>
+#include <fstream>
+
+using namespace shogun;
+
+TEST(Cereal, Json_SGVector_load_equals_saved)
+{
+	const index_t size = 5;
+	SGVector<float64_t> a(size), b;
+	a.range_fill(1.0);
+
+	{
+		std::ofstream os("cerealtest_sgvector.cereal");
+		cereal::JSONOutputArchive archive(os);
+		archive(a);
+	}
+
+	{
+		std::ifstream is("cerealtest_sgvector.cereal");
+		cereal::JSONInputArchive archive(is);
+		archive(b);
+	}
+	for (index_t i = 0; i < size; i++)
+		EXPECT_NEAR(a[i], b[i], 1E-15);
+}
+
+TEST(Cereal, Json_SGVector_load_equals_saved_refcounting_false)
```

Comment:
i would add a test that creates a vector call `SG_REF` on it 2-3 times, serialize it and then check the deserialized vectors ref_count value...